pkg/server/router/manifest.go

```diff
@@ -196,33 +208,78 @@ func (mr ManifestRouter) DeleteManifest(ctx context.Context, w http.ResponseWrit
 }
 
 type SubmitApplicationRequest struct {
-	// Once we have JWT signed wrapper that can get the did this can be removed
-	ApplicantDID string                            `json:"applicantDid" validate:"required"`
-	Application  manifestsdk.CredentialApplication `json:"credential_application" validate:"required"`
-	Credentials  []interface{}                     `json:"verifiableCredentials" validate:"required"`
+	ApplicationJWT keyaccess.JWT `json:"applicationJwt" validate:"required"`
+	// Contains the following properties:
+	// Application  manifestsdk.CredentialApplication `json:"credential_application" validate:"required"`
+	// Credentials  []interface{}                     `json:"vcs" validate:"required"`
 }
 
+const (
+	vcsJSONProperty                   = "vcs"
+	verifiableCredentialsJSONProperty = "verifiableCredentials"
+)
+
 func (sar SubmitApplicationRequest) ToServiceRequest() (*manifest.SubmitApplicationRequest, error) {
-	credContainer, err := credential.NewCredentialContainerFromArray(sar.Credentials)
+	parsed, err := jwt.Parse([]byte(sar.ApplicationJWT))
+	if err != nil {
+		return nil, errors.Wrap(err, "could not parse Credential Application token")
+	}
+
+	// make sure the known properties are present (Application and Credentials)
+	kid, ok := parsed.Get(jwk.KeyIDKey)
+	if !ok || kid == "" {
+		return nil, errors.New("Credential Application token missing kid")
+	}
+	var creds []interface{}
+	credentials, ok := parsed.Get(vcsJSONProperty)
+	if !ok {
+		logrus.Warn("could not find vc in Credential Application token, looking for `verifiableCredentials`")
+		if credentials, ok = parsed.Get(verifiableCredentialsJSONProperty); !ok {
+			return nil, errors.New("could not find vc or verifiableCredentials in Credential Application token")
+		}
+	}
+	creds = credentials.([]interface{})
```

Comment:
This can actually crash horiblly if it doesn't cast correctly, I think the 8 line json marshal casting thing will give err if something goes wrong

Reply:
will do a check here